framework/unSupervisedLearning.py

```diff
@@ -220,9 +220,25 @@ def train(self, tdict, metric = None):
         self.normValues = np.zeros((cardinality,cardinality))
         keys = tdictNorm.keys()
         for i in range(cardinality):
-          for j in range(i+1,cardinality):
-            self.normValues[i][j] = metric.distance(tdictNorm[keys[i]],tdictNorm[keys[j]])
-            self.normValues[j][i] = self.normValues[i][j]
+          for j in range(i,cardinality):
```

Comment:
these loops will be super slow. You do not have another way?

Reply:
I do add ```FIXME``` in line 238. This is an current issue in our devel branch. I do not have a good solution for this now. I would like to explore this issue in the future PR when I started to add more metrics. This may be changed since I plan to use the data object directly instead of dictionary here.